src/natcap/invest/habitat_quality.py

```diff
@@ -904,23 +1011,159 @@ def make_linear_decay_kernel_path(max_distance, kernel_path):
+                args['threats_table_path'], 'THREAT', to_lower=True).items()}
+        sensitivity_dict = utils.build_lookup_from_csv(
+            args['sensitivity_table_path'], 'LULC', to_lower=True)
+
+        # check that the threat names in the threats table match with the
+        # threats columns in the sensitivity table.
+        sens_header_set = set(list(sensitivity_dict.values())[0])
+        threat_set = {threat for threat in threat_dict}
+        missing_sens_header_set = threat_set.difference(sens_header_set)
+
+        if missing_sens_header_set:
+            validation_warnings.append(
+                (['sensitivity_table_path'],
+                 (f'Threats "{missing_sens_header_set}" does not match any'
+                  ' column in the sensitivity table. Sensitivity columns:'
+                  f' {sens_header_set}')))
+
+            invalid_keys.add('snsitivity_table_path')
+
+        # Get the directory path for the Threats CSV, used for locating threat
+        # rasters, which are relative to this path
+        threat_csv_dirpath = os.path.dirname(args['threats_table_path'])
+
+        # Validate threat raster paths and their nodata values
+        bad_threat_paths = []
+        duplicate_paths = []
+        threat_path_list = []
+        for lulc_key, lulc_args in (('_c', 'lulc_cur_path'),
+                                    ('_f', 'lulc_fut_path'),
+                                    ('_b', 'lulc_bas_path')):
+            if lulc_args in args:
+                # for each threat given in the CSV file try opening the
+                # associated raster which should be found in
+                # threat_raster_folder
+                for threat in threat_dict:
+                    # Threat path from threat CSV is relative to CSV
+                    threat_path = os.path.join(
+                        threat_csv_dirpath,
+                        threat_dict[threat][_THREAT_SCENARIO_MAP[lulc_key]])
+
+                    # check gis type of threat path and catch thrown ValueError
+                    # from get_gis_type if path does not exist
+                    try:
+                        threat_gis_type = pygeoprocessing.get_gis_type(
+                            threat_path)
+                        # if threat path not of type RASTER then raise
+                        # valueerror
+                        if threat_gis_type != pygeoprocessing.RASTER_TYPE:
+                            raise ValueError
+                    except ValueError:
+                        # it's okay to have no threat raster for baseline
+                        # scenario
+                        if lulc_key != '_b':
+                            bad_threat_paths.append(
+                                    (threat, _THREAT_SCENARIO_MAP[lulc_key]))
+                            continue
+                        else:
+                            threat_path = None
```

Comment:
This is similar to that block I commented on above, would you consider using a standard if/else instead of raising an exception just to catch it?

Reply:
Fixed here as mentioned above: https://github.com/natcap/invest/pull/75/commits/36fd0d9aa4f02357bab8bb5d47c26b7b03d8998f